src/Object.jl

```diff
@@ -0,0 +1,37 @@
+# MIT license
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# See LICENSE in the project root for full license information.
+
+using Base: uniontypes
+using Random
+using UUIDs
+
+export Object, Properties, getobjectproperty
+
+uuid() = uuid1(Random.GLOBAL_RNG)
+
+struct Object{T}
+    object::T
+    id::UUID
+
+    Object(object::T, id::UUID = uuid()) where {T} = new{T}(object, id)
+end
+
+Base.show(io::IO, object::Object) = print(io, "[$(object.id)]")
+
+unionobject(T::Type) = Union{T, Object{<:T}}
```

Comment:
I don't think the compiler will be able to reason about types that result from a function statically - I think using macros for this purpose would work better

Reply:
Although seems ok given that only the planar shapes test is causing allocs